packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -117,7 +117,99 @@ class TweenAnimationBuilder<T extends Object?> extends ImplicitlyAnimatedWidget
+  /// rotations.
+  ///
+  /// {@template flutter.widgets.TweenAnimationBuilder.repeat.reverse}
+  /// The [reverse] parameter controls the animation direction:
+  ///  * When `false` (default), the animation repeats in one direction, jumping
+  ///    back to the beginning when it completes.
+  ///  * When `true`, the animation reverses direction when it reaches the end,
+  ///    creating a back-and-forth motion.
+  /// {@endtemplate}
+  ///
+  /// {@template flutter.widgets.TweenAnimationBuilder.repeat.paused}
+  /// The [paused] parameter controls whether the animation is running:
+  ///  * When `false` (default), the animation runs continuously.
+  ///  * When `true`, the animation pauses at its current value.
+  ///  * Changing from `true` to `false` resumes the animation from where it paused.
+  /// {@endtemplate}
+  ///
+  /// Unlike the default constructor, [onEnd] is not supported for repeating
+  /// animations since they never end.
+  ///
+  /// {@tool snippet}
+  /// This example shows a spinning square that rotates continuously:
+  ///
+  /// ```dart
+  /// TweenAnimationBuilder<double>.repeat(
+  ///   tween: Tween<double>(begin: 0, end: 1),
+  ///   duration: const Duration(seconds: 2),
+  ///   builder: (BuildContext context, double value, Widget? child) {
+  ///     return Transform.rotate(
+  ///       angle: value * 2 * math.pi,
+  ///       child: Container(
+  ///         width: 100,
+  ///         height: 100,
+  ///         color: Colors.green,
+  ///       ),
+  ///     );
+  ///   },
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// {@tool snippet}
+  /// This example shows a pulsing effect using the [reverse] parameter:
+  ///
+  /// ```dart
+  /// TweenAnimationBuilder<double>.repeat(
+  ///   tween: Tween<double>(begin: 0.8, end: 1.2),
+  ///   duration: const Duration(seconds: 1),
+  ///   reverse: true,
+  ///   builder: (BuildContext context, double value, Widget? child) {
+  ///     return Transform.scale(
+  ///       scale: value,
+  ///       child: child,
+  ///     );
+  ///   },
+  ///   child: const Icon(Icons.favorite, color: Colors.red, size: 100),
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [AnimationController.repeat], which this constructor replaces for simple use cases.
+  ///  * [TweenAnimationBuilder], the default constructor for single animations.
+  ///
+  /// ## Ownership
+  ///
+  /// The [TweenAnimationBuilder] takes full ownership of the provided [tween]
+  /// instance and mutates it. Once a [Tween] has been passed to a
+  /// [TweenAnimationBuilder], its properties should not be accessed or changed
+  /// anymore to avoid interference with the [TweenAnimationBuilder].
+  const TweenAnimationBuilder.repeat({
+    super.key,
+    required this.tween,
+    required super.duration,
+    super.curve = Curves.linear,
+    bool reverse = false,
+    bool paused = false,
+    required this.builder,
+    this.child,
+  }) : _isRepeating = true,
```

Comment:
Oof.. This is not as nice as I had hoped. Maybe this would be better as a separate `RepeatedTweenAnimationBuilder` widget?